### ISSUE TITLE
Linter terminal application

### ISSUE THREAD

**Issue by FLuzzi-csw** [integrated-application-development/delphilint#75]
### Prerequisites

- [X] This improvement has not already been suggested.
- [X] This improvement would be generally useful, not specific to my code or setup.
- [X] This improvement is not related to analysis quality (e.g. new rules, improvements to existing rules), which should be raised on the [SonarDelphi repository](https://github.com/integrated-application-development/sonar-delphi/issues).

### Engine area

Other

### Improvement description

A CLI implementation of the linter could be really useful during CI.
This could enable users to have an automated and more direct feedback inside the workflow's output (think of it like `eslint`).

What do you think?
If this idea is out of the scope of the project feel free to close this issue

Thanks for you work!

### Rationale

Settings up CI infrastructure to have automated linting is pretty tedious and expensive.
Having option to simply install/download the `jar`, pass it the `Delphi` installation folder and just analyze could be really useful (given that the output is readable enough :smile:).

Example usage:

```shell
java delphilint.jar --delphi "C:\path\to\delphi" --rules "path\to\delphilint.ini" --sonar-host "http://..." --sonar-token "..." lint path/to/project
# give the option to read the "sonar values" from environment variables
# --rules could be optional, search `delphilint.ini` in the project directory (like `.sonar-project.properties`)
```

Example GH action workflow:

```yaml
name: Lint

on:
  pull_request:
    branches:
      - dev
      - main

# kill workflows from the same source
concurrency:
  group: ${{ github.workflow_ref }}-${{ github.event_name }}
  cancel-in-progress: true

jobs:
  lint:
    runs-on: [ self-hosted, delphi ]
    # if application is a jar, install and configure Java
    - name: Set up Java
      uses: actions/setup-java@v3
      with:
        distribution: 'temurin'
        java-version: '11'
    - name: Lint code
      uses: integrated-application-development/use-delphilint@v1
      with:
        project_dir: project/dir # default "."
        delphi_home: ${{ SECRETS.DELPHI_HOME }} #required
        rules: path/to/rule/file #optional
        sonar_host: ${{ SECRETS.SONAR_HOST }} #optional
        sonar_login: ${{ SECRETS.SONAR_LOGIN }} #optional
```

**Comment by fourls** [comment]
Hi @FLuzzi-csw,

A CLI for DelphiLint is out of scope of this project, since DelphiLint is just an IDE plugin that brings [SonarDelphi](https://github.com/integrated-application-development/sonar-delphi) (a Delphi language plugin for the SonarQube code analysis platform) into the Delphi IDE. It sounds like you might be looking for something closer to SonarDelphi here.

The idea with SonarDelphi is that you set up a SonarQube server (the [Community Build](https://www.sonarsource.com/open-source-editions/sonarqube-community-edition/) is free and open-source), which then provides a long-term home for code quality issues in your project. As a plugin, SonarDelphi is designed to run in CI via the sonar-scanner command line tool, which uploads the issues (and other metrics, like lines of code, test coverage, etc.) to your SonarQube server. 

While I haven't personally used SonarDelphi in a GitHub workflow, it should be doable by setting up a public SonarQube server and using the [Official SonarQube Scan](https://github.com/marketplace/actions/official-sonarqube-scan) action.

For more information I would recommend reading the SonarDelphi [README](https://github.com/integrated-application-development/sonar-delphi) and [manual](https://github.com/integrated-application-development/sonar-delphi/blob/master/docs/MANUAL.md).

Hopefully that answers your question!
